x-pack/plugins/watcher/public/sections/watch_edit/components/watch_visualization.tsx

```diff
@@ -93,28 +92,32 @@ const getTimeBuckets = (watch: any) => {
   return timeBuckets;
 };
 
-const loadWatchVisualizationData = debounce(async (watch: any, setWatchVisualizationData: any) => {
-  const domain = getDomain(watch);
-  const timeBuckets = new TimeBuckets();
-  timeBuckets.setBounds(domain);
-  const interval = timeBuckets.getInterval().expression;
-  const visualizeOptions = new VisualizeOptions({
-    rangeFrom: domain.min,
-    rangeTo: domain.max,
-    interval,
-    timezone: getTimezone(),
-  });
-  const { visualizeData } = (await getWatchVisualizationData(watch, visualizeOptions)) as any;
-  setWatchVisualizationData(visualizeData || {});
-}, 500);
-
 const WatchVisualizationUi = () => {
   const { watch } = useContext(WatchContext);
   const [watchVisualizationData, setWatchVisualizationData] = useState<any>({});
-
+  const [isLoading, setIsLoading] = useState(false);
+  const loadWatchVisualizationData = async () => {
+    const domain = getDomain(watch);
+    const timeBuckets = new TimeBuckets();
+    timeBuckets.setBounds(domain);
+    const interval = timeBuckets.getInterval().expression;
+    const visualizeOptions = new VisualizeOptions({
+      rangeFrom: domain.min,
+      rangeTo: domain.max,
+      interval,
+      timezone: getTimezone(),
+    });
+    setIsLoading(true);
+    const { visualizeData } = (await getWatchVisualizationData(watch, visualizeOptions)) as any;
```

Comment:
could you update `getWatchVisualizationData` in `api.ts` to use the `use_request` helper CJ added? I _think_ that returns an `isLoading` prop.

Reply:
Got an error when I tried to do this.  Punting for now.